x-pack/plugins/ml/public/embeddables/anomaly_charts/use_anomaly_charts_input_resolver.ts

```diff
@@ -81,9 +81,11 @@ export function useAnomalyChartsInputResolver(
 
           anomalyExplorerService.setTimeRange(timeRangeInput);
 
-          let influencersFilterQuery: InfluencersFilterQuery;
+          let influencersFilterQuery: InfluencersFilterQuery | undefined;
           try {
-            influencersFilterQuery = processFilters(filters, query);
+            if (filters && query) {
```

Comment:
Can there be a case where there are filters but no query or vice versa? If yes suggestion to change this to an `or` condition.

Reply:
In the context of the dashboard, there are both defined, and the code that processes filter and query assumes that it is defined at all times. Updated the condition [here](https://github.com/elastic/kibana/pull/139628/commits/3d78d515233bf16241fe40533b931b3873f1218a), but later on, the logic for processing filters and query might need revisiting.